multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -493,6 +566,221 @@ test.serial('distribute FastUSDC contract fees', async t => {
+  );
+  t.log('got forwardingAddress', userForwardingAddr);
+
+  const evidence = makeFakeEvidence(
+    mintAmt,
+    userForwardingAddr,
+    recipientAddress,
+  );
+
+  const queryClient = makeQueryClient(
+    await useChain(eudChain).getRestEndpoint(),
+  );
+
+  t.log(`UX->${eudChain}`, 'minting before evidence observed');
+  nobleTools.mockCctpMint(mintAmt, userForwardingAddr);
+  // Wait for mint to complete before submitting evidence
+  await sleep(5000, { log: t.log, setTimeout });
+
+  // submit evidences
+  await Promise.all(txOracles.map(o => o.submit(evidence)));
+  t.log(`UX->${eudChain}`, 'submitted x', txOracles.length);
+
+  await assertTxStatus(evidence.txHash, 'FORWARDED');
+  await t.notThrowsAsync(async () => {
+    await retryUntilCondition(
+      () => queryClient.queryBalance(EUD, getUsdcDenom(eudChain)),
+      ({ balance }) => {
+        if (!balance) return false; // retry
+        const value = BigInt(balance.amount);
+        if (value === 0n) return false; // retry
+        if (value < mintAmt) {
+          throw Error(`fees were deducted: ${value} < ${mintAmt}`);
+        }
+        t.log('forward done', value, 'uusdc');
+        return true;
+      },
+      `${EUD} forward available from fast-usdc`,
+      // this resolves quickly, so _decrease_ the interval so the timing is more apparent
+      { retryIntervalMs: 500, maxRetries: 20 },
+    );
+  });
+});
+
+test.serial('insufficient LP funds and forward failed', async t => {
+  const mintAmt = LP_DEPOSIT_AMOUNT * 2n;
+  const {
+    assertTxStatus,
+    makeFakeEvidence,
+    nobleTools,
+    nobleAgoricChannelId,
+    txOracles,
+    vstorageClient,
+  } = t.context;
+
+  // EUD wallet on the specified chain
+  const eudWallet = await createWallet('invalideud');
+  const EUD = (await eudWallet.getAccounts())[0].address;
+  t.log(`EUD wallet created: ${EUD}`);
+
+  // parameterize agoric address
+  const { settlementAccount } = await vstorageClient.queryData(
+    `published.${contractName}`,
+  );
+  t.log('settlementAccount address', settlementAccount);
+
+  const recipientAddress = encodeAddressHook(settlementAccount, { EUD });
+  t.log('recipientAddress', recipientAddress);
+
+  // register forwarding address on noble
+  const txRes = nobleTools.registerForwardingAcct(
+    nobleAgoricChannelId,
+    recipientAddress,
+  );
+  t.is(txRes?.code, 0, 'registered forwarding account');
+
+  const { address: userForwardingAddr } = nobleTools.queryForwardingAddress(
+    nobleAgoricChannelId,
+    recipientAddress,
+  );
+  t.log('got forwardingAddress', userForwardingAddr);
+
+  const evidence = makeFakeEvidence(
+    mintAmt,
+    userForwardingAddr,
+    recipientAddress,
+  );
+
+  // submit evidences
+  await Promise.all(txOracles.map(async o => o.submit(evidence)));
+
+  await assertTxStatus(evidence.txHash, 'OBSERVED');
+
+  nobleTools.mockCctpMint(mintAmt, userForwardingAddr);
+
+  await assertTxStatus(evidence.txHash, 'FORWARD_FAILED');
+});
```

Comment:
For the `ForwardFailed` tests, it might be worth showing that the `tx.amount` is sitting in the SettlementAccount's bank balance. The serial tests maybe make that tricky so I'd suggest capturing balance at the beginning of the test and compare it.